docarray/utils/_internal/misc.py

```diff
@@ -52,6 +52,10 @@
     'pymilvus': '"docarray[milvus]"',
 }
 
+ProtocolType = Literal[
```

Comment:
Followed the `ALLOWED_PROTOCOLS` constant at https://github.com/docarray/docarray/blob/main/docarray/array/doc_list/io.py#L54C1-L54C1
 
 It makes me uncomfortable to define the same information twice, so I can investigate combining the `ProtocolType` and `ALLOWED_PROTOCOLS` objects, even though one is a type object and the other is a set of strings.

Reply:
In a follow-up PR you can define ALLOWED_PROTOCOLS in relation to these types